Consul/Helper/MultiSemaphore.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace SensioLabs\Consul\Helper;
+
+use SensioLabs\Consul\Services\KV;
+use SensioLabs\Consul\Services\Session;
+use SensioLabs\Consul\Helper\MultiSemaphore\Resource;
+
+class MultiSemaphore implements MultiSemaphoreInterface
+{
+    /**
+     * @var Session
+     */
+    private $session;
+
+    /**
+     * @var KV
+     */
+    private $kv;
+
+    /**
+     * @var string
+     */
+    private $sessionId;
+
+    /**
+     * @var array
+     */
+    private $resources;
+
+    /**
+     * @var string
+     */
+    private $keyPrefix;
+
+    /**
+     * @var integer
+     */
+    private $ttl;
+
+    /**
+     * @var string
+     */
+    private $metaDataKey = '.semaphore';
+
+    /**
+     * @param Resource[] $resources
+     * @param int        $ttl
+     * @param Session    $session
+     * @param KV         $kv
+     * @param string     $keyPrefix
+     */
+    public function __construct(array $resources, $ttl, Session $session, KV $kv, $keyPrefix)
+    {
+        $this->resources = $resources;
+        $this->ttl = $ttl;
+        $this->session = $session;
+        $this->kv = $kv;
+        $this->keyPrefix = trim($keyPrefix, '/');
+    }
+
+    /**
+     * @return Resource[]
+     */
+    public function getResources()
+    {
+        return $this->resources;
+    }
+
+    /**
+     * @return bool|mixed
+     */
+    public function acquire()
+    {
+        if ($this->sessionId) {
+            throw new \Exception('Resources are acquired.');
+        }
+
+        $result = true;
+
+        // Start a session
+        $session = $this->session->create('{"Name": "semaphor", "LockDelay":0, "TTL": "'.$this->ttl.'s"}')->json();
+        $this->sessionId = $session['ID'];
+
+        $lockedResources = [];
+        foreach ($this->resources as $resource) {
+            if (false === $this->kv->put($this->getResourceKey($resource, $this->sessionId), '', ['acquire' => $this->sessionId])->json()) {
+                $result = false;
+            } else {
```

Comment:
There is another case in else block when i set false to result
And for me it is better to have only one break command inside foreach instead of heaving 2 break commands.